client/my-sites/plugins/plugin-management-v2/hooks/use-plugin-version-info.ts

```diff
@@ -0,0 +1,82 @@
+import { useMemo } from 'react';
+import { siteObjectsToSiteIds } from 'calypso/my-sites/plugins/utils';
+import { useSelector } from 'calypso/state';
+import { getPluginOnSites } from 'calypso/state/plugins/installed/selectors';
+import getSites from 'calypso/state/selectors/get-sites';
+import type { PluginComponentProps } from '../types';
+import type { SiteDetails } from '@automattic/data-stores';
+
+export default function usePluginVersionInfo(
+	plugin: PluginComponentProps,
+	selectedSite?: SiteDetails
```

Comment:
Since we're only ever interested in the selected site's ID, would it be more appropriate to pass only the ID itself? This would also satisfy the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter), since we would no longer be requesting extraneous data.

Reply:
Agree. I have made the changes: https://github.com/Automattic/wp-calypso/pull/81712/commits/cc6a3a7ec3d1a8a0ae67b8c5ff08634995b27da6